### DISCUSSION TITLE
Migrating from v2.5 to v3 in Next.js with Typescript

### DISCUSSION THREAD

**Discussion by glend1** [djipco/webmidi#206] [Migration]
A lot of stuff seems to be broken for me, I hope you don't mind helping me through this. Fixes might have a cascading effect so rather than detailing every problem I would prefer to work them out one by one.

````
import { WebMidi } from 'webmidi';
....
WebMidi.enable().then(() => {
    for(var input in WebMidi.inputs) {
        ....
    }
})
....
````

here `WebMidi.inputs` is resolving to `any[]`
this is a problem because input is being cast to a `string` and then nothing can be done with it (as far as typescript is concerned)
what i am trying to do is add a listener for both of the `connected` and `disconnected` events that were present in 2.5

**Comment by glend1** [comment]
I also get this error when I try to run `npm run dev`

````
./node_modules/webmidi/dist/esm/webmidi.esm.min.js:450:352
Module not found: Can't resolve 'perf_hooks'
````

**Reply by djipco** [reply]
I think I know what the problem is. This is probably due to a fix I had to introduce recently for WebPack users. I just published a new version (3.0.4) to NPM can you check if it solves the problem?

**Comment by djipco** [comment]
I had a hard time finding TypeScript users during the testing phase so I can definitely help you out through this if you don't mind feeding me a little...

I'm in a meeting right now but I can get back to you later.

**Comment by glend1** [comment]
absolutely, i want this fixed as much as you do.

**Reply by djipco** [reply]
Somebody else had the same issue. I updated the definition file. Can you try it out and tell me if it fixes the problem? Details here: https://github.com/djipco/webmidi/issues/204#issuecomment-995205847

**Comment by glend1** [comment]
Turns out I forgot what my code was doing when I decided to migrate I am actually trying to access the event that is triggered when devices are connected or disconnected. However, im still debugging to resolve any issues I might have.

**Comment by djipco** [comment]
Version 3.0.9 includes an updated version of the TypeScript declaration file which resolve outstanding issues with TypeScript.
